test/resources/tektonresults.go

```diff
@@ -0,0 +1,144 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	mfc "github.com/manifestival/client-go-client"
+	mf "github.com/manifestival/manifestival"
+
+	"knative.dev/pkg/test/logging"
+
+	"github.com/tektoncd/operator/test/utils"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	resultv1alpha1 "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EnsureTektonResultExists creates a TektonResult with the name names.TektonResult, if it does not exist.
+func EnsureTektonResultExists(clients resultv1alpha1.TektonResultInterface, names utils.ResourceNames) (*v1alpha1.TektonResult, error) {
+	// If this function is called by the upgrade tests, we only create the custom resource, if it does not exist.
+	trCR, err := clients.Get(context.TODO(), names.TektonResult, metav1.GetOptions{})
+	if err == nil {
+		return trCR, err
+	}
+	if apierrs.IsNotFound(err) {
+		trCR = &v1alpha1.TektonResult{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: names.TektonResult,
+			},
+			Spec: v1alpha1.TektonResultSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					TargetNamespace: names.TargetNamespace,
+				},
+			},
+		}
+		return clients.Create(context.TODO(), trCR, metav1.CreateOptions{})
+	}
+	return trCR, err
+}
+
+// WaitForTektonResultState polls the status of the TektonResult called name
+// from client every `interval` until `inState` returns `true` indicating it
+// is done, returns an error or timeout.
+func WaitForTektonResultState(clients resultv1alpha1.TektonResultInterface, name string,
+	inState func(s *v1alpha1.TektonResult, err error) (bool, error)) (*v1alpha1.TektonResult, error) {
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForTektonResultState/%s/%s", name, "TektonResultIsReady"))
+	defer span.End()
+
+	var lastState *v1alpha1.TektonResult
+	waitErr := wait.PollImmediate(Interval, Timeout, func() (bool, error) {
+		lastState, err := clients.Get(context.TODO(), name, metav1.GetOptions{})
+		return inState(lastState, err)
+	})
+
+	if waitErr != nil {
+		return lastState, fmt.Errorf("tektonresult %s is not in desired state, got: %+v: %w", name, lastState, waitErr)
+	}
+	return lastState, nil
+}
+
+// IsTektonResultReady will check the status conditions of the TektonResult and return true if the TektonResult is ready.
+func IsTektonResultReady(s *v1alpha1.TektonResult, err error) (bool, error) {
+	return s.Status.IsReady(), err
+}
+
+// AssertTektonResultCRReadyStatus verifies if the TektonResult reaches the READY status.
+func AssertTektonResultCRReadyStatus(t *testing.T, clients *utils.Clients, names utils.ResourceNames) {
```

Comment:
it's a bit odd to me to have an exported func in a library that requires a `testing.T`. Can we unexport these and move them to the test file where they are used?

Reply:
oh yeah make sense. It seems the for all components it is similar way. I will address this in a separate pr